GitTfs.VsCommon/TfsHelper.Common.cs

```diff
@@ -85,20 +88,31 @@ public void EnsureAuthenticated()
                     uri = new Uri(Url);
                 }
 
-                // TODO: Use TfsTeamProjectCollection constructor that takes a TfsClientCredentials object
+#if VS2010
```

Comment:
>Other version-specific stuff uses subclasses or composition to vary the behavior. 

This PR was also intended to remove all the warning during the build of the Git-Tfs.Vs201x projects (to build ALL the git-tfs projects without warning!) and this is the only way to achieve it, by preventing that the deprecated api appears in these projects (and also guarantee for the future that the projects will build even if the api is removed!).

I also was not very fond of `#if` but I think that's the better solution in this specific case...